firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/messages.txt

```diff
@@ -0,0 +1,207 @@
+/* Write function
+Scenario: Message: The write method correctly generates an encoded body from the protobuf message descriptor payload
+
+  GIVEN("A Message object default-constructed with StateSegment Taggedunion, default message types and a max size of 252 bytes as its class template parameters")
+  // assertions carried out in given:
+  // message class type member is 0
+
+    // setup:
+    // message class type member set to 5
+    // payload of parameters_request is set
+    // message descriptor array containing one element corresponding to Unknown message
+
+    WHEN("Write method is called on a message object with it's value of the payload tag greater than the size of the descriptor array")
+      THEN("The write method reports invalid type status")
+      THEN("After the write method is called, the type member remains unchanged")
+      THEN("The payload.values field data remains unchanged")
+      THEN("The output buffer remains empty")
+
+    // setup:
+    // message class type member is unchanged
+    // payload tag is set to Unknown
+
+    WHEN("Write method is called on a message object where it's payload tag value is such that it corresponds to Unrecognized message descriptor in descriptor array")
+      THEN("The write method reports invalid type status")
+      THEN("After the write method is called, the type member remains unchanged")
+      THEN("The payload.values field data remains unchanged")
+      THEN("The output buffer remains empty")
+
+  GIVEN("A Message object default-constructed with StateSegment Taggedunion, default message types and a max size of 126 bytes as its class template parameters")
+    // setup:
+    // sensor_measurements.flow = 30;
+    // sensor_measurements.fio2 = 85;
+    // sensor_measurements.spo2 = 72;  
+    WHEN("The payload is a sensor measurements message and write method is called")
+      THEN("The write method reports ok status")
+      THEN("After the write method is called, the type member is set to the value of the message payload tag")
```

Comment:
Does the `payload.value` field remain unchanged too? (This applies to all tests of the write method)

Reply:
fixed.